Can't commit due to an error in lint in git hooks

Output error:

```
main.go:31:56: Top level comment should end in a period (godot)
// given a template, and a config, generate shell script
                                                       ^
source.go:7:40: Top level comment should end in a period (godot)
// ugh, needs to be turned into a config
                                       ^
treewalk.go:16:42: Top level comment should end in a period (godot)
// TreeConfig is the project configuration
                                         ^
main.go:240: File is not `gofumpt`-ed (gofumpt)
	var ctx = context.New(*project)
main.go:292: File is not `gofumpt`-ed (gofumpt)

main.go:310:12: G306: Expect WriteFile permissions to be 0600 or less (gosec)
		if err = ioutil.WriteFile(*output, out, 0666); err != nil {
		         ^
treewalk.go:155:13: G306: Expect WriteFile permissions to be 0600 or less (gosec)
			if err = ioutil.WriteFile(shellpath, shellcode, 0644); err != nil {
			         ^
main.go:124:14: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"name_template %q contains unknown conditional or ARM format. Please file bug at https://github.com/goreleaser/godownloader\", target)" (goerr113)
		return "", fmt.Errorf("name_template %q contains unknown conditional or ARM format. Please file bug at https://github.com/goreleaser/godownloader", target)
		           ^
main.go:178:14: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"could not fetch a goreleaser configuration file\")" (goerr113)
	return nil, fmt.Errorf("could not fetch a goreleaser configuration file")
	            ^
main.go:209:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"repo or file not specified\")" (goerr113)
		return nil, fmt.Errorf("repo or file not specified")
		            ^
main.go:229:16: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"owner/name repo not specified\")" (goerr113)
			return nil, fmt.Errorf("owner/name repo not specified")
			            ^
shell_equinoxio.go:14:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"must have repo\")" (goerr113)
		return nil, fmt.Errorf("must have repo")
		            ^
shell_godownloader.go:10:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unable to parse: %s\", err)" (goerr113)
		return nil, fmt.Errorf("unable to parse: %s", err)
		            ^
shell_godownloader.go:23:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unable generate archive name: %s\", err)" (goerr113)
		return nil, fmt.Errorf("unable generate archive name: %s", err)
		            ^
shell_godownloader.go:29:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unable generate checksum name: %s\", err)" (goerr113)
		return nil, fmt.Errorf("unable generate checksum name: %s", err)
		            ^
shell_raw.go:14:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"must have GitHub owner/repo\")" (goerr113)
		return nil, fmt.Errorf("must have GitHub owner/repo")
		            ^
source.go:23:15: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unknown source %q\", source)" (goerr113)
		return nil, fmt.Errorf("unknown source %q", source)
		            ^
treewalk.go:91:11: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"invalid path: %s\", path)" (goerr113)
			return fmt.Errorf("invalid path: %s", path)
			       ^
treewalk.go:99:11: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"only github.com supported, got %s\", org)" (goerr113)
			return fmt.Errorf("only github.com supported, got %s", org)
			       ^
main.go:187:24: mnd: Magic number: 200, in <condition> detected (gomnd)
	if resp.StatusCode != 200 {
	                      ^
treewalk.go:90:20: mnd: Magic number: 3, in <condition> detected (gomnd)
		if len(parts) != 3 {
		                 ^
main.go:56:2: return with no blank line before (nlreturn)
	return out.Bytes(), err
	^
main.go:65:2: return with no blank line before (nlreturn)
	return platform
	^
main.go:97:2: return with no blank line before (nlreturn)
	return platformBinaries
	^
main.go:183:23: net/http.Get must not be called (noctx)
	resp, err := http.Get(file)
	                     ^
treewalk.go:33:2: directive `// nolint: gosec` is unused for linter gosec (nolintlint)
	// nolint: gosec
	^
treewalk.go:61:69: directive `// nolint: gocyclo,gocognit` is unused for linter gocyclo (nolintlint)
func treewalk(root string, treeout string, forceWrite bool) error { // nolint: gocyclo,gocognit
```

--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ fmt: ## gofmt and goimports all go files
 	find . -name '*.go' -not -wholename './vendor/*' | while read -r file; do gofmt -w -s "$$file"; goimports -w "$$file"; done
 
 lint: ## Run all the linters
-	./bin/golangci-lint run --enable-all --disable wsl ./...
+	./bin/golangci-lint run --enable-all --disable wsl,goerr113 ./...
 
 precommit: lint  ## Run precommit hook
 

--- a/main.go
+++ b/main.go
@@ -184,7 +184,7 @@ func loadURL(file string) (*config.Project, error) {
 	if err != nil {
 		return nil, err
 	}
-	if resp.StatusCode != 200 {
+	if resp.StatusCode != http.StatusOK {
 		log.Errorf("reading %s returned %d %s\n", file, resp.StatusCode, http.StatusText(resp.StatusCode))
 		return nil, nil
 	}

--- a/main.go
+++ b/main.go
@@ -302,7 +302,7 @@ func main() {
 	// only write out if forced to, OR if output is effectively different
 	// than what the file has.
 	if *force || shell.ShouldWriteFile(*output, out) {
-		if err = ioutil.WriteFile(*output, out, 0666); err != nil {
+		if err = ioutil.WriteFile(*output, out, 0600); err != nil {
 			log.WithError(err).Errorf("unable to write to %s", *output)
 			os.Exit(1)
 		}

--- a/main.go
+++ b/main.go
@@ -179,7 +179,7 @@ func loadURLs(path, configPath string) (*config.Project, error) {
 }
 
 func loadURL(file string) (*config.Project, error) {
-	// nolint: gosec
+	// nolint: gosec,noctx
 	resp, err := http.Get(file)
 	if err != nil {
 		return nil, err

--- a/main.go
+++ b/main.go
@@ -271,11 +271,11 @@ func main() {
 	log.SetHandler(cli.Default)
 
 	var (
-		repo    = kingpin.Flag("repo", "owner/name or URL of GitHub repository").Short('r').String()
-		output  = kingpin.Flag("output", "output file, default stdout").Short('o').String()
-		force   = kingpin.Flag("force", "force writing of output").Short('f').Bool()
-		tree    = kingpin.Flag("tree", "use tree to generate multiple outputs").String()
-		file    = kingpin.Arg("file", "??").String()
+		repo   = kingpin.Flag("repo", "owner/name or URL of GitHub repository").Short('r').String()
+		output = kingpin.Flag("output", "output file, default stdout").Short('o').String()
+		force  = kingpin.Flag("force", "force writing of output").Short('f').Bool()
+		tree   = kingpin.Flag("tree", "use tree to generate multiple outputs").String()
+		file   = kingpin.Arg("file", "??").String()
 	)
 
 	kingpin.CommandLine.Version(fmt.Sprintf("%v, commit %v, built at %v", version, commit, datestr))

--- a/main.go
+++ b/main.go
@@ -28,7 +28,7 @@ var (
 	datestr = "unknown"
 )
 
-// given a template, and a config, generate shell script
+// given a template, and a config, generate shell script.
 func makeShell(tplsrc string, cfg *config.Project) ([]byte, error) {
 	// if we want to add a timestamp in the templates this
 	//  function will generate it

--- a/main.go
+++ b/main.go
@@ -53,6 +53,7 @@ func makeShell(tplsrc string, cfg *config.Project) ([]byte, error) {
 		return nil, err
 	}
 	err = t.Execute(&out, cfg)
+
 	return out.Bytes(), err
 }
 
@@ -62,6 +63,7 @@ func makePlatform(goos, goarch, goarm string) string {
 	if goarch == "arm" && goarm != "" {
 		platform += "v" + goarm
 	}
+
 	return platform
 }
 
@@ -94,6 +96,7 @@ func makePlatformBinaries(cfg *config.Project) map[string][]string {
 			}
 		}
 	}
+
 	return platformBinaries
 }
 
@@ -142,6 +145,7 @@ func makeName(prefix, target string) (string, error) {
 		return "", err
 	}
 	err = t.Execute(&out, varmap)
+
 	return out.String(), err
 }
 
@@ -175,6 +179,7 @@ func loadURLs(path, configPath string) (*config.Project, error) {
 			return project, nil
 		}
 	}
+
 	return nil, fmt.Errorf("could not fetch a goreleaser configuration file")
 }
 
@@ -186,6 +191,7 @@ func loadURL(file string) (*config.Project, error) {
 	}
 	if resp.StatusCode != http.StatusOK {
 		log.Errorf("reading %s returned %d %s\n", file, resp.StatusCode, http.StatusText(resp.StatusCode))
+
 		return nil, nil
 	}
 	p, err := config.LoadReader(resp.Body)
@@ -195,11 +201,13 @@ func loadURL(file string) (*config.Project, error) {
 	if errc != nil {
 		return nil, errc
 	}
+
 	return &p, err
 }
 
 func loadFile(file string) (*config.Project, error) {
 	p, err := config.Load(file)
+
 	return &p, err
 }
 
@@ -281,6 +289,7 @@ func main() {
 			log.WithError(err).Error("treewalker failed")
 			os.Exit(1)
 		}
+
 		return
 	}
 
@@ -296,6 +305,7 @@ func main() {
 			log.WithError(err).Error("unable to write")
 			os.Exit(1)
 		}
+
 		return
 	}
 
@@ -306,6 +316,7 @@ func main() {
 			log.WithError(err).Errorf("unable to write to %s", *output)
 			os.Exit(1)
 		}
+
 		return
 	}
 

--- a/main.go
+++ b/main.go
@@ -237,7 +237,7 @@ func Load(repo, configPath, file string) (project *config.Project, err error) {
 		project.Dockers[i].Files = []string{}
 	}
 
-	var ctx = context.New(*project)
+	ctx := context.New(*project)
 	for _, defaulter := range defaults.Defaulters {
 		log.Infof("setting defaults for %s", defaulter)
 		if err := defaulter.Default(ctx); err != nil {
@@ -285,7 +285,6 @@ func main() {
 	}
 
 	out, err := processGodownloader(*repo, "", *file)
-
 	if err != nil {
 		log.WithError(err).Error("failed")
 		os.Exit(1)

--- a/treewalk.go
+++ b/treewalk.go
@@ -13,7 +13,7 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
-// TreeConfig is the project configuration
+// TreeConfig is the project configuration.
 type TreeConfig struct {
 	// these can be set by config
 	Source  string `yaml:"source,omitempty"`  // type of downloader to make
@@ -26,7 +26,7 @@ type TreeConfig struct {
 	name  string // repo name
 }
 
-// LoadTreeConfig Loads config file
+// LoadTreeConfig Loads config file.
 func LoadTreeConfig(file string) (config TreeConfig, err error) {
 	// nolint: gosec
 	f, err := os.Open(file)
@@ -37,7 +37,7 @@ func LoadTreeConfig(file string) (config TreeConfig, err error) {
 	return LoadTreeConfigReader(f)
 }
 
-// LoadTreeConfigReader config via io.Reader
+// LoadTreeConfigReader config via io.Reader.
 func LoadTreeConfigReader(fd io.Reader) (config TreeConfig, err error) {
 	data, err := ioutil.ReadAll(fd)
 	if err != nil {

--- a/treewalk.go
+++ b/treewalk.go
@@ -84,7 +84,7 @@ func treewalk(root string, treeout string, forceWrite bool) error { // nolint: g
 		// Now: github.com/owner/repo
 		// better way of doing this?
 		parts := strings.Split(rel, string(os.PathSeparator))
-		if len(parts) != 3 {
+		if len(parts) != 3 { // nolint: gomnd
 			return fmt.Errorf("invalid path: %s", path)
 		}
 

--- a/treewalk.go
+++ b/treewalk.go
@@ -144,7 +144,7 @@ func treewalk(root string, treeout string, forceWrite bool) error { // nolint: g
 		// only write out if forced to, OR if output is effectively different
 		// than what the file has.
 		if forceWrite || shell.ShouldWriteFile(shellpath, shellcode) {
-			if err = ioutil.WriteFile(shellpath, shellcode, 0644); err != nil {
+			if err = ioutil.WriteFile(shellpath, shellcode, 0600); err != nil {
 				return err
 			}
 		}

--- a/treewalk.go
+++ b/treewalk.go
@@ -28,7 +28,6 @@ type TreeConfig struct {
 
 // LoadTreeConfig Loads config file.
 func LoadTreeConfig(file string) (config TreeConfig, err error) {
-	// nolint: gosec
 	f, err := os.Open(file)
 	if err != nil {
 		return
@@ -56,7 +55,7 @@ func LoadTreeConfigReader(fd io.Reader) (config TreeConfig, err error) {
 // https://github.com/goreleaser/godownloader/issues/64
 //
 // nolint: funlen
-func treewalk(root string, treeout string, forceWrite bool) error { // nolint: gocyclo,gocognit
+func treewalk(root string, treeout string, forceWrite bool) error { // nolint: gocognit
 	rooterr := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
 		// weird case where filewalk failed
 		if err != nil {

--- a/treewalk.go
+++ b/treewalk.go
@@ -33,6 +33,7 @@ func LoadTreeConfig(file string) (config TreeConfig, err error) {
 		return
 	}
 	log.WithField("file", file).Debug("loading config file")
+
 	return LoadTreeConfigReader(f)
 }
 
@@ -44,6 +45,7 @@ func LoadTreeConfigReader(fd io.Reader) (config TreeConfig, err error) {
 	}
 	err = yaml.UnmarshalStrict(data, &config)
 	log.WithField("config", config).Debug("loaded config file")
+
 	return config, err
 }
 
@@ -110,6 +112,7 @@ func treewalk(root string, treeout string, forceWrite bool) error { // nolint: g
 		}
 		if conf.Ignore {
 			log.WithField("repo", rel).Warn("ignoring repo as instructed")
+
 			return nil
 		}
 
@@ -152,5 +155,6 @@ func treewalk(root string, treeout string, forceWrite bool) error { // nolint: g
 		// we did it!
 		return nil
 	})
+
 	return rooterr
 }

--- a/treewalk.go
+++ b/treewalk.go
@@ -16,8 +16,8 @@ import (
 // TreeConfig is the project configuration.
 type TreeConfig struct {
 	// these can be set by config
-	Source  string `yaml:"source,omitempty"`  // type of downloader to make
-	Config  string `yaml:"config,omitempty"`  // sets a custom location for goreleaser.yml config file
+	Source string `yaml:"source,omitempty"` // type of downloader to make
+	Config string `yaml:"config,omitempty"` // sets a custom location for goreleaser.yml config file
 
 	// these can not be set by config file
 	// and are set by the url/path

